### ISSUE TITLE
Exposing header via cors-expose-header Issue

### ISSUE THREAD

**Issue by xoraingroup** [tus/tusd#1059]
**Question**
I have tried to expose x-token using -cors-allow-headers x-token -cors-expose-headers x-token , but I am not able to get it in pre-flight response header.

<img width="1468" alt="image" src="https://github.com/tus/tusd/assets/17129052/8256bf06-a063-4ede-a897-edce290333de">

Tried all possible combinations and used following branches and tags

main branch
v2.2.0 
v2.2.1

I am not able to get those headers. Am I doing it all wrong? Please help. Thank you

**Setup details**
Please provide following details, if applicable to your situation:
- Operating System: [e.g. Linux, Windows, macOS]

Linux

- Used tusd version: [can be obtained by running `tusd -version`]
For some weird reason it shows NA 
```
[root@minio woopsie]# tusd --version
Version: n/a
Commit: n/a
Date: n/a
```

Last commit of the main branch is 
```
commit e2784199a268bd03cdc188f376386fadc6735582 (HEAD -> main, origin/main, origin/HEAD)
Author: Martin Stemmle <martnst@users.noreply.github.com>
Date:   Wed Dec 20 09:14:28 2023 +0100

    docs: Unify style of CLI arguments (#1050)

    Update CLI option examples from 2 leading dashes to just 1.
```

- Used tusd data storage: [e.g. disk store, AWS S3, Google Cloud Storage]
Minio Storage S3

- Used tusd configuration: [what arguments are you using to invoke `tusd`?]
```
tusd -host 127.0.0.1 -port 1080 -base-path=/resumable -s3-bucket mybucket -s3-object-prefix / -s3-disable-ssl true -cors-allow-headers x-token -cors-expose-headers x-token -hooks-enabled-events pre-create,post-create,post-finish -hooks-http http://localhost:3002 -hooks-http-retry 10 -hooks-http-backoff 2 -hooks-stop-code 401 -behind-proxy -expose-metrics
```

- Used tus client library: [e.g. tus-js-client, tus-java-client, TUSKit, Uppy.js]
Uppy.js

**Comment by Acconut** [comment]
> * For some weird reason it shows NA

The version is only available if you use the pre-built binaries. If you compile it on your own, no version information is embedded in the output. If you were using the pre-built binary and still didn't, get any version details, you might be running into the bug reported in https://github.com/tus/tusd/issues/1045 and fixed in the latest v2.2.1 release.

> I am not able to get those headers. Am I doing it all wrong? Please help. Thank you

I am not able to reproduce this. Running tusd via `tusd -cors-allow-headers x-token -cors-expose-headers x-token`, the response from curl incluses the header:

```
~ $ curl -X OPTIONS -i http://[::]:8080/files/ -H "Origin: example.com"
HTTP/1.1 200 OK
Access-Control-Allow-Headers: Authorization, Origin, X-Requested-With, X-Request-ID, X-HTTP-Method-Override, Content-Type, Upload-Length, Upload-Offset, Tus-Resumable, Upload-Metadata, Upload-Defer-Length, Upload-Concat, Upload-Complete, Upload-Draft-Interop-Version, x-token
Access-Control-Allow-Methods: POST, HEAD, PATCH, OPTIONS, GET, DELETE
Access-Control-Allow-Origin: example.com
Access-Control-Max-Age: 86400
Tus-Extension: creation,creation-with-upload,termination,concatenation,creation-defer-length
Tus-Resumable: 1.0.0
Tus-Version: 1.0.0
Vary: Origin
X-Content-Type-Options: nosniff
Date: Mon, 08 Jan 2024 14:18:12 GMT
Content-Length: 0
```

In Firefox, I can also see the value in the preflight response:

![image](https://github.com/tus/tusd/assets/1375043/141bbfab-6dc9-4efb-9a11-daae91fe92b4)

**Comment by Acconut** [comment]
I am closing this since I assume that the question has been answered. Feel free to continue here if that's not the case.
